src/payment/strategies/amazon-maxo/amazon-maxo-payment-processor.ts

```diff
@@ -0,0 +1,57 @@
+import { CheckoutStore } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+
+import { AmazonMaxoButtonParams, AmazonMaxoSDK } from './amazon-maxo';
+import AmazonMaxoScriptLoader from './amazon-maxo-script-loader';
+
+export default class AmazonMaxoPaymentProcessor {
+    private _amazonMaxoSDK?: AmazonMaxoSDK;
+    private _methodId?: string;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _amazonMaxoScriptLoader: AmazonMaxoScriptLoader
+    ) { }
+
+    initialize(methodId: string): Promise<void> {
+        this._methodId = methodId;
+
+        return this._configureWallet();
+    }
+
+    createButton(containerId: string, params: AmazonMaxoButtonParams): HTMLElement {
+        if (!this._amazonMaxoSDK) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._amazonMaxoSDK.Pay.renderButton(containerId, params);
+    }
+
+    private _configureWallet(): Promise<void> {
+        const methodId = this._getMethodId();
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                return this._amazonMaxoScriptLoader.load(paymentMethod)
+                    .then(amazonMaxoClient => {
```

Comment:
there seems to be a bit too much nesting here. Maybe change to async/await syntax to improve readability?

```js
const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);

if (!paymentMethod) {
  throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
}

const amazonMaxoClient = await this._amazonMaxoScriptLoader.load(paymentMethod);
// ...
```

Reply:
Async/await syntax implemented.